### DISCUSSION TITLE
CSI snapshot support

### DISCUSSION THREAD

**Discussion by hopoffbaby** [weka/csi-wekafs#143] [Q&A] [answered]
Hi all,

With the recent updates allowing CSI snapshots to be taken from kubernetes, I'd like to understand which StorageClass configurations are supported:

> Starting from CSI Plugin v2.0, three StorageClass configurations are available:
Directory-backed StorageClass 
Snapshot-backed StorageClass
Filesystem-backed StorageClass

The documentation states weka has a limit of 1024 file systems. For a large k8s deployment that is sharing a weka cluster with other workloads this will quickly become a problem if 1 filesystem == 1 PV. 

From my understanding weka snapshotting happens at the filesystem level.

Is this CSI driver enabling snapshots of PVs backed by weka directories therefore working around this filesystem limitation?

Thanks

**Comment by sergeyberezansky** [comment, marked as answer]
Hello,
The question is also answered here: https://github.com/weka/csi-wekafs/issues/43

Weka does have a limit of 1024 filesystems. However, the limit of snapshots (read-only and writable ones) is significantly higher.

Directory-backed volumes utilize a single shared filesystem,
Filesystem-backed volumes utilize a filesystem per volume, and of course are limited by the max. supported filesystems
Snapshot-backed volumes utilize a writable snapshot of dedicated, empty filesystem and help to overcome the limitation. 
Please refer to Weka documentation for the particular version of cluster you work with for exact limitations on number of snapshots
